docs/configuration/configuration.md

```diff
@@ -963,8 +963,12 @@ once the labels are removed.
 ### `<metric_relabel_configs>`
 
 Metric relabeling is applied to samples as the last step before ingestion. It
-has the same configuration format and actions as target relabeling. Metric
-relabeling does not apply to automatically generated timeseries such as `up`.
+has the same configuration format and actions as target relabeling. However,
+the labels starting with `__` are not exposed to the `<metric_relabel_configs>`
```

Comment:
I am not sure this is accurate. The `__meta` labels are removed at the end of the `relabel_configs` but other `__` prefixed labels are present.

Reply:
Yes, as James says these labels never get to this stage. This is documented in the relabel_configs docs.

Reply:
You appear to be correct if you refer to that code and comment: https://github.com/prometheus/prometheus/blob/master/retrieval/target.go#L320

I actually managed to misread the second flow diagram from that blog post to arrive at this conclusion.